[merge AFTER transpile tab goes live] Add transpile to start/index page (#421)
Qiskit/documentation: docs/start/index.mdx

- [Build](../build): Design and develop quantum circuits with primitives and advanced methods like dynamic circuits and mid-circuit measurements. You can also find our circuit library here.

> - [Transpile](../transpile): Compile locally to transform your circuits to be hardware-executable, with varying degrees of error awareness.

Author: ```suggestion
- [Transpile](../transpile): Compile locally to optimize your circuits to run efficiently on hardware, with varying degrees of error awareness.
```
